### PR TITLE
fix RHEL CI warning for rosbag2_storage_mcap.

### PR DESCRIPTION
**Pull request by fujitatomoya**
see RHEL failure https://ci.ros2.org/job/ci_linux-rhel/1801/testReport/rosbag2_storage_mcap/

### CODE REVIEW

**Review by fujitatomoya — COMMENTED** [review]
@MichaelOrlov can you take a look?

**Review by fujitatomoya — COMMENTED** [review]
i am in the middle of RHEL CI https://ci.ros2.org/job/ci_linux-rhel/1801/testReport/rosbag2_storage_mcap/ and https://github.com/ros2/rosbag2/actions/runs/12334336949/job/34424404138?pr=1883... 😓 any thoughts?

**Comment by MichaelOrlov** [comment]
@fujitatomoya I am bit confused. Do you see this error on Rolling or Humble? And only on RHEL or other distros too?
If this is on Humble, it is a bit unexpected and suspicious that `ament_calng_format` was taking from rolling. From your log.
```
Setting up ros-rolling-ament-clang-format (0.11.4-1focal.20220120.172447) ...
/usr/bin/bash -c source /opt/ros/rolling/setup.sh && ament_clang_format $(colcon list --packages-select rosbag2_storage_mcap -p) --config rosbag2_storage_mcap/.clang-format
Code style divergence in file 'rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_message_definition_cache.cpp':
```

**Comment by MichaelOrlov** [comment]
@fujitatomoya I have a suspicious that `$(colcon list --packages-select rosbag2_storage_mcap -p)` fails to find `rosbag2_storage_macp` package for some reasons and `ament_clang_fromat` ignores local config from the `rosbag2_storage_mcap/.clang-format` and trying to check against default clang_format configuration.

**Comment by MichaelOrlov** [comment]
See in the log https://ci.ros2.org/job/ci_linux-rhel/1801/consoleText
```
rhel/ws/build/rosbag2_storage_mcap/ament_clang_format/clang_format.txt" "--command" "/home/jenkins-agent/workspace/ci_linux-rhel/ws/install/ament_clang_format/bin/ament_clang_format" "--xunit-file" "/home/jenkins-agent/workspace/ci_linux-rhel/ws/build/rosbag2_storage_mcap/test_results/rosbag2_storage_mcap/clang_format.xunit.xml" "--config" ".clang-format"
```
As far as I understand the clang format trying to refer to the the default `.clang-format`  file. which is not going to be present in the build folder at least. And therefore clang-format silently fallback to the default `.clang-format` configuration.

Need to compare how the same clang format check running on Rolling CI or how it was running before it starts failig.

**Comment by fujitatomoya** [comment]
> Do you see this error on Rolling or Humble?

Humble only.

> And only on RHEL or other distros too?

Yeah, RHEL only.

> it is a bit unexpected and suspicious that ament_calng_format was taking from rolling. From your log.

So this is really weird... why does it not use `humble` version for that... this does not look like a problem in rosbag2.

**Comment by clalancette** [comment]
> @fujitatomoya I am bit confused. Do you see this error on Rolling or Humble? And only on RHEL or other distros too?
> If this is on Humble, it is a bit unexpected and suspicious that `ament_calng_format` was taking from rolling. From your log.

This is a bug in the GitHub workflow on Humble: https://github.com/ros2/rosbag2/blob/humble/.github/workflows/lint.yml#L21 .  I'll submit a PR to fix it.

**Comment by fujitatomoya** [comment]
@clalancette thanks! i was taking a look at the same line! happy to review PR!

**Comment by clalancette** [comment]
See #1884, which should fix this.

**Comment by fujitatomoya** [comment]
Test CI with `rosbag2_storage_mcap` to see if clang_format still sees the failure. (current code is the same with humble head)

**Comment by fujitatomoya** [comment]
Pulls: ros2/rosbag2#1883
Gist: https://gist.githubusercontent.com/fujitatomoya/22ddf66f8e25e92464b14c5cfa549390/raw/95fa7054ffaefd956248eb5a80a8bc0f9ac23798/ros2.repos
BUILD args:  --packages-up-to rosbag2_storage_mcap
TEST args:  --packages-select rosbag2_storage_mcap
ROS Distro: humble
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14973
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/22287/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16547/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1814/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/23076/)

File: rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_message_definition_cache.cpp [hunk 1]
```diff
@@ -151,10 +151,14 @@ TEST(test_message_definition_cache, get_service_message_definitions)
 TEST(test_local_message_definition_source, throw_definition_not_found_for_unknown_msg)
 {
   MessageDefinitionCache source;
-  ASSERT_THROW({ source.get_full_text("rosbag2_test_msgdefs/msg/UnknownMessage"); },
-               rosbag2_storage_mcap::internal::DefinitionNotFoundError);
```
Comment (fujitatomoya) [inline rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_message_definition_cache.cpp]: https://ci.ros2.org/job/ci_linux-rhel/1814/testReport/ can be reproducible with RHEL CI.

and this PR fixes the format aligned with rolling one.

**Comment by fujitatomoya** [comment]
- `rosbag2_storage_mcap/.clang-format` is exactly same between `rolling` and `humble`.
- humble original RHEL CI: https://ci.ros2.org/job/ci_linux-rhel/1814/ fails without this PR. (this is original issue that i am trying to address)
- rolling REHL CI: https://ci.ros2.org/job/ci_linux-rhel/1820/ succeeds.
- if we change this into being aligned with rolling format: github workflow still fails https://ci.ros2.org/job/ci_linux-rhel/1821/ even after rebase https://github.com/ros2/rosbag2/pull/1884.

the other thing is that github workflow uses released package, and CI builds the package for `ament-clang-format`.
i tried using both `ros-humble-ament-clang-format` and source code build, it does not change the outcome.

after all, github workflow seems to be fine to me by local verification.

something could be wrong with RHEL CI but the following CI commands are the same between rolling and humble.

```bash
Test command: /usr/bin/python3 "-u" "/home/jenkins-agent/workspace/ci_linux-rhel/ws/install/ament_cmake_test/share/ament_cmake_test/cmake/run_test.py" "/home/jenkins-agent/workspace/ci_linux-rhel/ws/build/rosbag2_storage_mcap/test_results/rosbag2_storage_mcap/clang_format.xunit.xml" "--package-name" "rosbag2_storage_mcap" "--output-file" "/home/jenkins-agent/workspace/ci_linux-rhel/ws/build/rosbag2_storage_mcap/ament_clang_format/clang_format.txt" "--command" "/home/jenkins-agent/workspace/ci_linux-rhel/ws/install/ament_clang_format/bin/ament_clang_format" "--xunit-file" "/home/jenkins-agent/workspace/ci_linux-rhel/ws/build/rosbag2_storage_mcap/test_results/rosbag2_storage_mcap/clang_format.xunit.xml" "--config" ".clang-format"
```

i cannot think of anything else... any thoughts?

**Comment by clalancette** [comment]
> i cannot think of anything else... any thoughts?

My best guess of what is happening here is that the different versions of clang on Ubuntu 22.04 (1.14.0) and RHEL-8 (18.1.8) have different formatting.  We deal with this a lot when updating e.g. uncrustify.

The good news is that there is a solution which I believe will satisfy both.  I'm going to push a commit to this branch in a minute here to see if that is the case.

**Comment by clalancette** [comment]
All right, the only failure here was on the Hpr job, but that didn't have to do with formatting.  So I'll say that this is fixed.  I'm going to run CI on it.  @MichaelOrlov since I did some work here, I'd like your review on this.

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
Re-run of CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/22296/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/16556/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1827/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/23086/)

**Comment by fujitatomoya** [comment]
@clalancette thanks for checking this! hopefully CI goes green 🤞🤞🤞

**Comment by fujitatomoya** [comment]
@MichaelOrlov all green!

@clalancette thanks again 🚀  this took my time longer than expected.... small things take time 😅

**Review by MichaelOrlov — APPROVED** [review]

**Pull request merged**